src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,104 @@
+     * @throws IllegalDescriptionException If the description of the task is illegal.
+     * @throws IllegalCommandException If the command is illegal.
+     */
+    public Command parseCommand(String command)
+            throws IllegalDescriptionException, IllegalCommandException {
+        Command cmd;
+        if (command.equals(ExitCommand.COMMAND_WORD)) {
+            cmd = new ExitCommand();
+        } else if (command.equals(ListCommand.COMMAND_WORD)) {
+            cmd = new ListCommand();
+        } else {
+            int indexOfSpace = command.indexOf(' ');
+            //if there is no space, assume that the string is a command type
+            if (indexOfSpace == -1) {
+                indexOfSpace = command.length();
+            }
+            //seperate command and description of the task
+            String type = command.substring(0, indexOfSpace);
+            String description = command.substring(indexOfSpace).strip();
+
+            if (type.equals(DoneCommand.COMMAND_WORD)) {
+                cmd = new DoneCommand(Integer.parseInt(description));
+            } else if (type.equals(DeleteCommand.COMMAND_WORD)) {
+                cmd = new DeleteCommand(Integer.parseInt(description));
+            } else if (type.equals(ToDoCommand.COMMAND_WORD)) {
+                cmd = new ToDoCommand(description.strip());
+            } else if (type.equals(DeadlineCommand.COMMAND_WORD)) {
+                int sep = description.indexOf("/by");
+                if (sep == -1) {
+                    throw new IllegalDescriptionException("The format of deadline task is wrong.");
+                }
+                cmd = new DeadlineCommand(description.substring(0, sep).strip(),
+                                parseDate(description.substring(sep + 3).strip()));
+            } else if (type.equals(EventCommand.COMMAND_WORD)) {
+                int sep = description.indexOf("/at");
+                if (sep == -1) {
+                    throw new IllegalDescriptionException("The format of event task is wrong.");
+                }
+                cmd = new EventCommand(description.substring(0, sep).strip(),
+                                parseDate(description.substring(sep + 3).strip()));
+            } else if (type.equals(FindCommand.COMMAND_WORD)) {
+                cmd = new FindCommand(description);
+            } else {
+                throw new IllegalCommandException(
+                        "I'm sorry, but I don't know what that means :-(");
+            }
+        }
+        return cmd;
+    }
+
+    /**
+     * Returns a LocalDateTime object representing the date and time of String date.
+     * @param date a String that is to be parsed into date and time
+     * @return a LocalDateTime object representing the date and time of String date.
+     * @throws IllegalDescriptionException If the date or time is illegal.
+     */
+    private LocalDateTime parseDate(String date) throws IllegalDescriptionException {
```

Comment:
Perhaps could include the correct date / time format for either the user, programmer (in comments) or both! It could possibly take some time to figure out the proper date format of the String to be parsed, even if the user has the code.